A1_perception:
  node:
    period_ms: 10
    inputs:
      robot_position:
        topic: "/amcl_pose"
        type: "PoseWithCovarianceStamped"
      one_d_tof:
        topic: "/sensor_to_pointcloud/tof/mono"
        type: "PointCloud2"
      multi_tof_left_row1:
        topic: "/sensor_to_pointcloud/tof/multi/left/row_1"
        type: "PointCloud2"
      multi_tof_left_row2:
        topic: "/sensor_to_pointcloud/tof/multi/left/row_2"
        type: "PointCloud2"
      multi_tof_left_row3:
        topic: "/sensor_to_pointcloud/tof/multi/left/row_3"
        type: "PointCloud2"
      multi_tof_left_row4:
        topic: "/sensor_to_pointcloud/tof/multi/left/row_4"
        type: "PointCloud2"
      multi_tof_right_row1:
        topic: "/sensor_to_pointcloud/tof/multi/right/row_1"
        type: "PointCloud2"
      multi_tof_right_row2:
        topic: "/sensor_to_pointcloud/tof/multi/right/row_2"
        type: "PointCloud2"
      multi_tof_right_row3:
        topic: "/sensor_to_pointcloud/tof/multi/right/row_3"
        type: "PointCloud2"
      multi_tof_right_row4:
        topic: "/sensor_to_pointcloud/tof/multi/right/row_4"
        type: "PointCloud2"
      camera_points:
        topic: "/sensor_to_pointcloud/camera_object"
        type: "PointCloud2"
      # lidar_points:
      #   # 에브리봇 코드 확인후 수정해야 함
      #   topic: "/scan"
      #   type: "LaserScan"

    layers:
      one_d_tof:
        inputs:
        - "one_d_tof"
        filter:
          compose:
            filters:
              timeout:
                timeout_milliseconds: 30000
              polygon_roi:
                use_inside: true
                polygon: [ [0.65, -0.65], [0.65, 0.65], [-1.0,1.0], [-1.0,-1.0]]
              polygon_roi:
                use_inside: false
                polygon: [ [0.18, -0.18], [0.18, 0.18], [0,0.18], [0,-0.18]]
              one_d_tof_stop:
                use_stop: true
      camera_points:
        inputs:
        - "camera_points"
        filter:
          compose:
            filters:
              timeout:
                timeout_milliseconds: 10000
              sector_roi:
                use_inside: true
                min_range: 0.0
                max_range: 1.0
                min_angle: -150
                max_angle: 150
              density:
                max_count: 3
                radius: 0.1
      multi_tof_12:
        inputs:
        - "multi_tof_left_row1"
        - "multi_tof_left_row2"
        - "multi_tof_right_row1"
        - "multi_tof_right_row2"
        filter:
          compose:
            filters:
              timeout:
                timeout_milliseconds: 15000
              sector_roi:
                use_inside: true
                min_range: 1.01
                max_range: 2.5
                min_angle: -160
                max_angle: 160
              density:
                max_count: 3
                radius: 0.1
      multi_tof_12_realtime:
        inputs:
        - "multi_tof_left_row1"
        - "multi_tof_left_row2"
        - "multi_tof_right_row1"
        - "multi_tof_right_row2"
        filter:
          compose:
            filters:
              timeout:
                timeout_milliseconds: 100
              sector_roi:
                use_inside: true
                min_range: 0.24
                max_range: 1.0
                min_angle: -50.3
                max_angle: 50.3
              density:
                max_count: 3
                radius: 0.1
      # multi_tof_3_left:
      #   inputs:
      #   - "multi_tof_left_row3"
      #   filter:
      #     compose:
      #       filters:
      #         drop_off:
      #           min_range: 0.59
      #           max_range: 0.7
      #           line_length: 0.05
      #           resolution: 0.05
      # multi_tof_3_right:
      #   inputs:
      #   - "multi_tof_right_row3"
      #   filter:
      #     compose:
      #       filters:
      #         drop_off:
      #           min_range: 0.61
      #           max_range: 0.7
      #           line_length: 0.05
      #           resolution: 0.05
                
      multi_tof_4_left:
        inputs:
        - "multi_tof_left_row4"
        filter:
          compose:
            filters:
              drop_off:
                min_range: 0.43
                max_range: 0.54
                line_length: 0.05
                resolution: 0.05
      multi_tof_4_right:
        inputs:
        - "multi_tof_right_row4"
        filter:
          compose:
            filters:
              drop_off:
                min_range: 0.44
                max_range: 0.50
                line_length: 0.05
                resolution: 0.05
